doc/xplat_cli_migration.md

```diff
@@ -0,0 +1,205 @@
+
+Once installed, you can run `az configure` and follow the steps to setup the
+Azure CLI 2.0 as well as help login.  Once this step is complete you should be
+to use both CLIs.  
+
+## Important new concepts in the Azure CLI 2.0
+Here is a quick list of some new and changed concepts that can help you understand the new tool.
+
+* Interactive Concepts
+  * Use `az configure` to setup your default output format
+  * You will find help to be generally more useful, try `az vm create -h` for an example
+  * Positional parameters are not supported, use `az vm list -g MyGroup` instead of `azure vm list MyGroup`
+* Automation and Scripting Concepts
+  * You can refer to resources using their full ID URI with `--ids` or with the resource group and name using `-g [rg] -n [name]`
+  * Use `--query "[expression]"` to extract single values
+  * Use `--out tsv` when getting a single value to prevent any extra markup
+  * Use `@-` to pipe values such as `az vm list --query [0].[id] --out tsv | az vm show --ids @-`
+* Service Specific Concepts
+  * VM power state is no longer included in `az vm list`, use `az vm get-instance-view` instead
+
+## Moving scripts from XPlat CLI to Azure CLI 2.0
+
+Generally, converting a script from XPlat to Az follows these steps:
+
+1. Switch `azure` commands to `az` commands
+2. Update commands to use new input values
+3. Update script to use new output formats
+4. Use `--query` to extract values reliably
+
+Below, we break down each of these steps.
+
+### Finding and switching to `az` commands
+
+While most commands keep the same group and command names between the Azure XPlat CLI and the Azure CLI 2.0, we've built a [azure to az conversion table](https://github.com/Azure/azure-cli/blob/master/doc/azure2az_commands.rst) for common commands.
+
+Mutate operations now use the `update` verb instead of `set`.  
+
+Some commands which required complex input, such as JSON strings and documents,
+have been changed to add/remove single values, instead of requiring users to
+pull down the document, modify, and set the new document.  For these commands,
+see help (`-h`) for details on the specific command.
+
+An example of this is `azure storage cors set` being replaced by `az storage cors add`.
+
+### Updating input values
+
+Once you have identified the `az` commands required for your script, you immediately
+notice changes to how inputs are handled.  The Azure CLI 2.0 does not accept
+'positional parameters' such as `azure vm show MyRG MyName`, but instead require
+parameter flags: `az vm show -g MyRG -n MyName`.  
+
+In addition, when an input value is missing, we will show an error indicating the
+missing parameters, instead of prompting the user automatically:
+
+```
+$ az vm show
+az vm show: error: (--name --resource-group | --ids) are required
+```
+
+In addition to use resource groups and names (`-g -n`), you can also refer to
+resources directly by ID value using `--ids`:
```

Comment:
It might be important to note here that the reason it's called **ids** is because you can pass a list of ids and it will loop the operation on each id.